packages/smart/contracts/turbo/AMMFactory.sol

```diff
@@ -133,7 +133,10 @@ contract AMMFactory is BNum {
         _collateral.transferFrom(msg.sender, address(this), _collateralIn);
         _collateral.approve(address(_marketFactory), MAX_UINT);
         uint256 _sets = _marketFactory.calcShares(_collateralIn);
-        _marketFactory.mintShares(_marketId, _sets, address(this));
+        uint256 _tokensIn = (_sets / _marketFactory.shareFactor()) * _marketFactory.shareFactor();
+
+        require(_tokensIn > 0, "Not enough collateral");
```

Comment:
Should be handled under the calcCost function mintShares calls, since the tokensIn has to be larger than shareFactor which is always larger than zero anyway.